install/helm/gloo/generate/values.go

```diff
@@ -606,6 +606,7 @@ type Mtls struct {
 	Enabled               *bool                 `json:"enabled,omitempty" desc:"Enables internal mtls authentication"`
 	Sds                   SdsContainer          `json:"sds,omitempty"`
 	EnvoySidecar          EnvoySidecarContainer `json:"envoy,omitempty"`
+	EnvoySidecarImage     *Image                `json:"envoySidecarImage,omitempty" desc:"Istio-proxy image to use for mTLS"`
```

Comment:
for the existing `EnvoySidecarContainer` above, it's used in the gloo deployment in the `envoy-sidecar` container.

how about to mirror that, we can name this new field IstioProxy
```suggestion
	IstioProxy     IstioProxyContainer               `json:"istioProxy,omitempty" desc:"Istio-proxy container"`
```
and define container:
```
type IstioProxyContainer struct {
	Image *Image `json:"image,omitempty" desc:"Istio-proxy image to use for mTLS"`
}
```

Reply:
@jenshu I updated the code according to your suggestion. However, I am not sure if you have seen my comment about the previous naming:

> it was dictated to me by the existing property https://github.com/solo-io/gloo/blob/master/install/helm/gloo/templates/7-gateway-proxy-deployment.yaml#L354-L356

With this update for someone to configure the istio-proxy container resources he would have to use the property: `envoySidecarResources` and to configure the image he'd have to use the `istioProxy.image`. Which are two different names envoy and istio proxy, while we mean istio-proxy in both cases.

Reply:
hmm, i think it's ok to keep the naming we have. in the gloo deployment, the envoySidecarResources is used under the `envoy-sidecar` container and the image uses `global.glooMtls.envoy.image`.
for this new one, i think `istioProxy.image` works because it matches the container name. it happens to reuse the envoySidecarResources value but i think that's ok. if we rename the envoySidecarResources in gateway-proxy-deployment, the user will have to define the resources in 2 places

Reply:
Sounds good to me!